File: gamedex-datamodel/src/main/java/com/github/ykrasik/gamedex/datamodel/persistence/ExcludedPath.java
package com.github.ykrasik.gamedex.datamodel.persistence;

import lombok.EqualsAndHashCode;
import lombok.NonNull;
import lombok.Value;

import java.nio.file.Path;

/**
 * @author Yevgeny Krasik
 */
@Value
@EqualsAndHashCode(of = "id")
public class ExcludedPath implements PathEntity {
    @NonNull private final Id<ExcludedPath> id;
    @NonNull private final Path path;

    @Override
    public String toString() {
        return path.toString();
    }
}

File: gamedex-datamodel/src/main/kotlin/com/github/ykrasik/gamedex/datamodel/persistence/ExcludedPath.kt
package com.github.ykrasik.gamedex.datamodel.persistence

import java.nio.file.Path

/**
 * User: ykrasik
 * Date: 25/05/2016
 * Time: 12:08
 */
data class ExcludedPath(
    val id: Int,
    val path: Path
) {
}